### ISSUE TITLE
Restoring an RDS instance from snapshot does not apply parameter groups

### ISSUE THREAD

**Issue by dynamike** [terraform-providers/terraform-provider-aws#1510]
When restoring an RDS instances from snapshot with a custom parameter group, the custom parameter group is not applied to the restored RDS instance and need to be rebooted for the changes to take effect.  

The AWS documentation [specifies](http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_RestoreFromSnapshot.html)

>When you restore a DB instance, the default DB parameter group is associated with the restored instance. As soon as the restore is complete and your new DB instance is available, you must associate any custom DB parameter group used by the instance you restored from. You must apply these changes by using the RDS console's Modify command, the ModifyDBInstance Amazon RDS API, or the AWS CLI modify-db-instance command.

Looking at the [code](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_db_instance.go#L516), ModifyDBInstance will get called if security groups or the password changes, but not the parameter group.

### Terraform Version
0.10.2

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_db_instance

### Terraform Configuration Files
```hcl
variable "aws_region" {
  default = "us-east-1"
}

provider "aws" {
  region = "${var.aws_region}"
}

resource "aws_db_subnet_group" "default" {
  name       = "main"
  subnet_ids = ["subnet-e45e14c8", "subnet-b8373ef0"]
}

resource "aws_db_parameter_group" "default" {
  name   = "default"
  family = "mysql5.6"

  parameter {
    name  = "character_set_server"
    value = "utf8"
  }
}

resource "aws_db_instance" "default" {
  apply_immediately = true

  allocated_storage    = 5
  storage_type         = "gp2"
  engine               = "mysql"
  engine_version       = "5.6.35"
  instance_class       = "db.m4.large"
  name                 = "mydb"
  username             = "foo"
  password             = "bar12345"
  db_subnet_group_name = "${aws_db_subnet_group.default.id}"
  parameter_group_name = "${aws_db_parameter_group.default.id}"
  snapshot_identifier  = "arn:aws:rds:us-east-1:889396472275:snapshot:instancepublic"
  skip_final_snapshot  = true
}
```
### Expected Behavior
I would expect the custom parameter group to report as being `in-sync` with the restored RDS instance. 

### Actual Behavior
Instead the restored RDS instance was reporting the parameter group as being 
![image](https://user-images.githubusercontent.com/675823/29729292-75c28a9e-8990-11e7-9e15-4e6aaf1f6f14.png)

### Steps to Reproduce
1. `terraform apply`
2. Verify the RDS instance parameter group is incorrectly showing as `pending-reboot`

**Comment by dynamike** [comment]
The more I look at the code and how RDS behaves, I realized that calling ModifyDBInstance won't do the trick. Terraform would need to explicitly call https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#RDS.RebootDBInstance to pick up the changes.

**Comment by bflad** [comment]
In version 1.33.0 of the AWS provider, likely releasing later today, the `aws_db_instance` will now correctly apply `parameter_group_name` on resource creation when using `snapshot_identifier`. Please note though that it will still be in `pending-reboot` state, which will also be fixed shortly.

**Comment by bflad** [comment]
The handling to prevent `pending-reboot` on creation (by issuing `RebootDBInstance`) has been merged into master and will release with version 1.34.0 of the AWS provider, likely later today.

**Comment by bflad** [comment]
This has been released in [version 1.34.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1340-august-30-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by hsharma96** [comment]
```
Terraform v0.11.10
provider.aws v1.50.0
```

I'm seeing the same issue in the latest terraform version. I'm trying to restore from a snapshot however it's not accepting the `parameter_group_name` and fails saying `Error creating DB Instance: InvalidParameterValue: You can't restore a DB instance with a custom ParameterGroup.` Has anyone been able to figure out this issue? 

Here's how my tf file looks like: 

```
provider "aws" {
    region = "us-gov-west-1"
}

# Get latest snapshot from production DB
data "aws_db_snapshot" "latest_snapshot" {
    most_recent             = true
    db_instance_identifier  = "non_prod_dev"
}

# Prints out the latest snapshot name
output "latest_snapshot" {
    value = "${data.aws_db_snapshot.latest_snapshot.id}"
}

variable "parameter_group_name" {
  default = "nonprod-postgres10-copy"
}

# Restore DB from the latest snapshot
resource "aws_db_instance" "latest_snapshot" {
    identifier                  = "testing"
    engine                      = "postgres"
    engine_version              = "10.3"
    license_model               = "postgresql-license"
    instance_class              = "db.m4.xlarge"
    storage_type                = "io1"
    iops                        = "5000"

    apply_immediately           = "true"

    username                    = "xxx"
    password                    = "xxx"

    db_subnet_group_name        = "xzy"
    publicly_accessible         = "true"

    parameter_group_name        = "${var.parameter_group_name}"
    option_group_name           = "default:postgres-10"
    vpc_security_group_ids      = ["xyz", "xzy"]

    auto_minor_version_upgrade  = "false"
    snapshot_identifier         = "${data.aws_db_snapshot.latest_snapshot.id}"
    skip_final_snapshot         = "true"

    tags = {
        Application     = "xyz"
        Environment     = "sbx"
        HasReplica      = "False"
        ReadReplica     = "False"
        Purpose         = "Testing"
    }

    timeouts = {
        create = "60m"
        delete = "2h"
    }
}
```

**Comment by bflad** [comment]
Hi @hsharma96 👋 Sorry for the trouble. Can you please open a separate issue for tracking? I'm guessing the RDS API update to migrate `DbParameterGroupName` from only being in `CreateDbInstance` to also being present in `CreateDbInstanceFromSnapshot` probably is not compatible with AWS GovCloud (US) yet. See also: https://github.com/terraform-providers/terraform-provider-aws/pull/6200 (released in version 1.41.0 of the AWS provider)
